src/nnbench/context.py

```diff
@@ -183,3 +187,256 @@ def __call__(self) -> dict[str, Any]:
         result["memory_unit"] = self.memunit
         # TODO: Lacks CPU cache info, which requires a solution other than psutil.
         return {self.key: result}
+
+
+class Context:
+    def __init__(self, context: dict[str, Any] | ContextProvider | "Context" | None = None) -> None:
+        self._ctx_dict: dict[str, Any] = {}
+        if context:
+            self.merge(context, inplace=True)
+
+    def keys(self) -> KeysView[str]:
+        return self._ctx_dict.keys()
+
+    def values(self) -> ValuesView[Any]:
+        return self._ctx_dict.values()
+
+    def items(self) -> ItemsView[str, Any]:
+        return self._ctx_dict.items()
+
+    def merge(
+        self,
+        other: "Context" | dict[str, Any] | ContextProvider,
+        this_ctx: dict[str, Any] | None = None,
+        inplace: bool = True,
+    ) -> "Context":
+        """
+        Merge another dictionary, Context, or ContextProvider into this Context.
+
+        Parameters
+        ----------
+        other : Context | dict[str, Any] | ContextProvider
+            The other context to merge into this one.
+        this_ctx : dict[str, Any] | None, optional
+            The target dictionary to merge into. Takes this classes context if None.
```

Comment:
This is redundant with `self` in OOP terms, so please leave it out.

(Alternatively: You probably would not want to merge two things in a class into a different object, that's functional programming.)

Reply:
We need the ability to pass a dict for the recursive call later. I am taking the same approach here as with the other methods (a private method working on dicts and an exposed instance method collectin the context dicts and passing them to the private one).